### ISSUE TITLE
Files marked as DnD prevent torrent status becoming completed

### ISSUE THREAD

**Issue by wopian** [jfurrow/flood#299]
Torrents where some files were marked as Do Not Download remain in the `Downloading` (green) status after reaching 100%, instead of changing to the `Completed` (blue) status.

![](http://i.imgur.com/6KgHubE.png)
![](http://i.imgur.com/4qQhmzd.png)

**Comment by jfurrow** [comment]
I agree, the torrent should be considered `completed` if all requested files have finished downloading.

This might be confusing for users, but I think the percentage of completion should always reflect the percentage of total data downloaded — e.g., if a user sets 50% of a torrent to `do not download`, then the torrent's status will be `completed` when the torrent's percentage reaches 50%. What do you think?

**Comment by wopian** [comment]
Looks like the related issues of https://github.com/rakshasa/rtorrent/issues/588 is somewhat relevant.
